src/codegen/function.rs

```diff
@@ -13,7 +13,7 @@ use super::translate_to_glib::TranslateToGlib;
 use writer::primitives::{format_block, tabs};
 use writer::ToCode;
 
-pub fn generate<W: Write>(w: &mut W, env: &Env, analysis: &analysis::functions::Info,
+pub fn generate(w: &mut Write, env: &Env, analysis: &analysis::functions::Info,
```

Comment:
It seems making `Untabber` be generic over the inner writer type would avoid the need to change the function signatures.